test/cases/base_test.rb

```diff
@@ -776,7 +776,11 @@ def test_collection_path
   def test_collection_path_with_parameters
     assert_equal "/people.json?gender=male", Person.collection_path(gender: "male")
     assert_equal "/people.json?gender=false", Person.collection_path(gender: false)
-    assert_equal "/people.json?gender=", Person.collection_path(gender: nil)
+    if ActiveSupport::VERSION::MAJOR < 8 || ActiveSupport::VERSION::MINOR < 1
```

Comment:
As an alternative to version checking, I also considered an explicit `{ key: nil }.to_query == "key"`, but felt that was too verbose and harder to `grep` for in the future when these branches are no longer necessary.